crates/bevy_pbr/src/render/gpu_preprocess.rs

```diff
@@ -0,0 +1,305 @@
+pub struct GpuPreprocessNode {
+    view_query: QueryState<(Entity, Read<PreprocessBindGroup>)>,
+}
+
+/// The compute shader pipeline for the mesh uniform building pass.
+#[derive(Resource)]
+pub struct PreprocessPipeline {
+    /// The single bind group layout for the compute shader.
+    pub bind_group_layout: BindGroupLayout,
+    /// The pipeline ID for the compute shader.
+    ///
+    /// This gets filled in in `prepare_preprocess_pipeline`.
+    pub pipeline_id: Option<CachedComputePipelineId>,
+}
+
+/// The compute shader bind group for the mesh uniform building pass.
+///
+/// This goes on the view.
+#[derive(Component)]
+pub struct PreprocessBindGroup(BindGroup);
+
+impl Plugin for GpuMeshPreprocessPlugin {
+    fn build(&self, app: &mut App) {
+        load_internal_asset!(
+            app,
+            MESH_PREPROCESS_SHADER_HANDLE,
+            "mesh_preprocess.wgsl",
+            Shader::from_wgsl
+        );
+
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.add_systems(
+            Render,
+            (
+                prepare_preprocess_pipeline.in_set(RenderSet::Prepare),
+                prepare_preprocess_bind_groups
+                    .run_if(
+                        resource_exists::<BatchedInstanceBuffers<MeshUniform, MeshInputUniform>>,
+                    )
+                    .in_set(RenderSet::PrepareBindGroups),
+            ),
+        );
+    }
+
+    fn finish(&self, app: &mut App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        // Stitch the node in.
+        render_app
+            .add_render_graph_node::<GpuPreprocessNode>(Core3d, NodePbr::GpuPreprocess)
+            .add_render_graph_edges(Core3d, (NodePbr::GpuPreprocess, Node3d::Prepass))
+            .init_resource::<PreprocessPipeline>()
+            .init_resource::<SpecializedComputePipelines<PreprocessPipeline>>();
+    }
+}
+
+impl FromWorld for GpuPreprocessNode {
+    fn from_world(world: &mut World) -> Self {
+        Self {
+            view_query: QueryState::new(world),
+        }
+    }
+}
+
+impl Node for GpuPreprocessNode {
+    fn update(&mut self, world: &mut World) {
+        self.view_query.update_archetypes(world);
+    }
+
+    fn run<'w>(
+        &self,
+        _: &mut RenderGraphContext,
+        render_context: &mut RenderContext<'w>,
+        world: &'w World,
+    ) -> Result<(), NodeRunError> {
+        // Grab the [`BatchedInstanceBuffers`]. If we aren't using GPU mesh
+        // uniform building, bail out.
+        let Some(BatchedInstanceBuffers {
+            work_item_buffers: ref index_buffers,
+            ..
+        }) = world.get_resource::<BatchedInstanceBuffers<MeshUniform, MeshInputUniform>>()
+        else {
+            error!(
+                "Attempted to preprocess meshes on GPU, but `GpuBuilt` batched instance buffers \
```

Comment:
`GpuBuilt`?

Reply:
Oops, that was from an older version of the patch. Fixing.